var/spack/repos/builtin/packages/paintor/package.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Paintor(MakefilePackage):
+    """Probabilistic Annotation integrator. Fast, integrative fine mapping with functional
+    data"""
+
+    homepage = "https://github.com/gkichaev/PAINTOR_V3.0"
+    url = "https://github.com/gkichaev/PAINTOR_V3.0/archive/refs/tags/3.0.tar.gz"
+
+    version("3.0", sha256="cc39d3c334cc6d787e4f04847192c9d0185025a2ca46910bd38901b6679d198f")
+
+    depends_on("nlopt")
+    depends_on("eigen")
+
+    def edit(self, spec, prefix):
+        makefile = FileFilter("Makefile")
+        makefile.filter("CC = g\+\+", "CC = {}".format(spack_cxx))
```

Comment:
You could also use an f-string:

```suggestion
        makefile.filter("CC = g\\+\\+", f"CC = {spack_cxx}")
```